openml/runs/run.py

```diff
@@ -106,6 +107,83 @@ def _generate_trace_arff_dict(self):
 
         return arff_dict
 
+    def get_metric_score(self, sklearn_fn, kwargs={}):
+        '''Calculates metric scores based on predicted values. Assumes the
+        run has been executed locally (and contans run_data). Furthermore,
+        it assumes that the 'correct' field has been set (which is
+        automatically the case for local runs)
+
+        Parameters
+        -------
+        sklearn_fn : function
+            a function pointer to a sklearn function that
+            accepts y_true, y_pred and *kwargs
+
+        Returns
+        -------
+        scores : list
+            a list of floats, of length num_folds * num_repeats
+        '''
+        if self.data_content is not None:
+            predictions_arff = self._generate_arff_dict()
+        elif 'predictions' in self.output_files:
+            raise ValueError('Not Implemented Yet: Function can currently only be used on locally executed runs (contributor needed!)')
+        else:
+            raise ValueError('Run should have been locally executed.')
+
+        def _attribute_list_to_dict(attribute_list):
+            # convenience function
+            res = dict()
+            for idx in range(len(attribute_list)):
+                res[attribute_list[idx][0]] = idx
+            return res
+        attribute_dict = _attribute_list_to_dict(predictions_arff['attributes'])
```

Comment:
It took my a while to figure out what you're doing here. Could you please add a comment like: `Create a mapping to map from the name of attributes present in the arff prediction file to their index. This is necessary because the number of classes can be different for different tasks.`?